modules/data.land/R/IC_SOILGRID_Utilities.R

```
@@ -0,0 +1,348 @@
+#' @param verbose Logical, print detailed progress information
+#' @param seed Optional random seed for reproducibility
+#' 
+#' @return Vector of soil carbon values with proper uncertainty handling
+#' @export
+generate_soilgrids_ensemble <- function(processed_data, site_id, size, depth_layer, verbose = FALSE, seed = NULL) {
+  if (verbose) {
+    PEcAn.logger::logger.info(sprintf("Generating %d ensemble members for site %s (%s)",size, site_id, depth_layer))
+  }
+  
+  if (!is.null(seed)) {
+    if (exists(".Random.seed", envir = .GlobalEnv)) {
+      old_seed <- .Random.seed
+      on.exit(assign(".Random.seed", old_seed, envir = .GlobalEnv))
+    }
+    set.seed(seed)
+  }
+  
+  site_row <- which(processed_data$data$Site_ID == site_id)
+  if (length(site_row) == 0) {
+    PEcAn.logger::logger.severe(sprintf("Site %s not found in processed data", site_id))
+  }
+  
+  # Select appropriate columns based on depth layer
+  if (depth_layer == "0-30cm") {
+    mean_c <- processed_data$data$`Total_soilC_0-30cm`[site_row]
+    original_sd <- processed_data$data$`Std_soilC_0-30cm`[site_row]
+    cv_dist <- processed_data$cv_distribution_30cm
+  } else {
+    mean_c <- processed_data$data$`Total_soilC_0-200cm`[site_row]
+    original_sd <- processed_data$data$`Std_soilC_0-200cm`[site_row]
+    cv_dist <- processed_data$cv_distribution_200cm
+  }
+  
+  if (is.na(mean_c) || mean_c <= 0) {
+    PEcAn.logger::logger.severe(sprintf("Invalid mean soil carbon value for site %s (%s)", 
+                                      site_id, depth_layer))
+  }
+  
+  soil_c_values <- numeric(size)
+  
+  # Use site-specific uncertainty
+  if (!is.na(original_sd) && original_sd > 0) {
+    shape <- (mean_c^2) / (original_sd^2)
+    rate <- mean_c / (original_sd^2)
+    if (is.finite(shape) && is.finite(rate) && shape > 0 && rate > 0) {
+      soil_c_values <- pmax(stats::rgamma(size, shape, rate), 0)
+    } else {
+      soil_c_values <- rep(mean_c, size)
```

Same comment as below: Do not fill means, return NA or throw an error. Silent fallbacks like this can badly distort the sampled distribution and leave the user unaware of potentially severe data problems.

---

fixed